test/check-application

```diff
@@ -2259,12 +2269,42 @@ class TestFiles(testlib.MachineCase):
                     b.click("button.pf-m-secondary:contains('Keep original')")
 
             b.wait_not_present(".pf-v5-c-modal-box")
-            b.wait_in_text(".pf-v5-c-alert__description", "Successfully uploaded file")
-            b.click(".pf-v5-c-alert__action button")
-            b.wait_not_present(".pf-v5-c-alert__action")
+            assert_upload_alert([os.path.basename(files[-1])], "admin:admin")
 
             self.assertEqual(m.execute(f"cat {dest_dir}/{filename}"), "new content")
 
+            # As administrator upload in testuser and change ownership to root:root
+
+            m.execute("useradd --create-home testuser")
+            b.go('/files#/?path=/home/testuser')
+            b.wait_not_present('.pf-v5-c-empty-state')
+
+            filename = os.path.basename(files[-1])
+            b.click("#upload-file-btn")
+            b.upload_files("#upload-file-btn + input[type='file']", [files[-1]])
+            assert_upload_alert([filename], "testuser:testuser", close=False)
+            self.assert_owner(f'/home/testuser/{filename}', 'testuser:testuser')
+
+            b.click(".pf-v5-c-alert__action-group button")
+            b.wait_not_present(".pf-v5-c-alert__action")
+            b.wait_in_text(".pf-v5-c-modal-box__title-text", filename)
+            # BUG BUG BUG gets all values but not the specific selected value
+            b.wait_val("#edit-permissions-owner", "testuser")
+            b.wait_val("#edit-permissions-group", "testuser")
+            b.wait_val("#edit-permissions-owner-access", "read-write")
+            b.wait_val("#edit-permissions-group-access", "read-only")
+            b.wait_val("#edit-permissions-other-access", "read-only")
+
+            b.select_from_dropdown("#edit-permissions-owner", "root")
+            b.select_from_dropdown("#edit-permissions-group", "root")
+            b.click("button.pf-m-primary")
+            b.wait_not_present(".pf-v5-c-modal-box")
+
+            self.assert_owner(f'/home/testuser/{filename}', 'root:root')
+
+            b.go(f'/files#/?path={dest_dir}')
+            b.wait_not_present('.pf-v5-c-empty-state')
```

Comment:
Please add a test that uploads/replaces an existing file. See above, this will need some more care, but we should already pin down  the behaviour (perhaps this will also just be denied, but then testing that is equally important).